### PR TITLE
Preserve channels-last memory format in to_image

### LINKED ISSUE #9547: Make `to_image` reading return tensors in CHW
### 🚀 The feature

Make to_image reading return tensors in CHW

### Motivation, pitch

https://github.com/pytorch/vision/blob/da446275eccae6cd00923080d9cc91f1f3a73744/torchvision/transforms/v2/functional/_type_conversion.py#L10-L25
```py
import pyvips
import torchvision,torchvision.transforms.v2

img = pyvips.Image.new_from_file(r"F:\test.jpg")
tensor = torchvision.transforms.v2.functional.to_image(img.numpy())
tensor.shape
#torch.Size([3, 1830, 1370])

tensor.shape, tensor.stride(), tensor.is_contiguous(memory_format=torch.channels_last), tensor.unsqueeze(0).is_contiguous(memory_format=torch.channels_last)

# (torch.Size([3, 1830, 1370]), (2507100, 1370, 1), False, False) # Not memory_format=torch.channels_last

new_tensor = torch.from_numpy(tensor.numpy()).permute((2,0,1))
new_tensor.shape, new_tensor.stride(), new_tensor.is_contiguous(memory_format=torch.channels_last), new_tensor.unsqueeze(0).is_contiguous(memory_format=torch.channels_last)

# (torch.Size([3, 1830, 1370]), (1, 4110, 3), False, True) # This is memory_format=torch.channels_last
```

> https://docs.pytorch.org/vision/main/transforms.html#range-and-dtype
...
Transforms tend to be sensitive to the input strides / memory format. Some transforms will be faster with channels-first images while others prefer channels-last. Like torch operators, most transforms will preserve the memory format of the input, but this may not always be respected due to implementation details. You may want to experiment a bit if you’re chasing the very best performance. Using [torch.compile()](https://docs.pytorch.org/docs/stable/generated/torch.compile.html#torch.compile) on individual transforms may also help factoring out the memory format variable (e.g. on [Normalize](https://docs.pytorch.org/vision/main/generated/torchvision.transforms.v2.Normalize.html#torchvision.transforms.v2.Normalize)). Note that we’re talking about memory format, not [tensor shape](https://docs.pytorch.org/vision/main/transforms.html#conventions).
>
> Note that resize transforms like [Resize](https://docs.pytorch.org/vision/main/generated/torchvision.transforms.v2.Resize.html#torchvision.transforms.v2.Resize) and [RandomResizedCrop](https://docs.pytorch.org/vision/main/generated/torchvision.transforms.v2.RandomResizedCrop.html#torchvision.transforms.v2.RandomResizedCrop) typically prefer channels-last input and tend not to benefit from [torch.compile()](https://docs.pytorch.org/docs/stable/generated/torch.compile.html#torch.compile) at this time.
...

As `torchvision.transforms.v2.functional.to_image` usually being the starting point of the processing pipeline, the latter process of it would always be `float().div_(255)` or `to_dtype` then `resize` and etc., these all prefer channels_last memory_format for better performance.
Therefore the starting point of the processing pipeline should keep its NHWC memory format. And which make the memcopy by `contiguous()` avoid that always happen (since image is always HWC when read in numpy)

Thanks for your reading.
I could make a PR

### PR DESCRIPTION
**Pull request by Hemanath400**
## Description

This PR avoids forcing a contiguous copy when converting a NumPy image to a torchvision `Image` in `to_image`.

Currently, the NumPy input path uses `.contiguous()` after `permute()`, which converts the tensor to a channels-first contiguous memory layout.

Removing `.contiguous()` preserves the original HWC memory layout while keeping the returned tensor in CHW shape, allowing the resulting tensor to be channels-last compatible.

## Changes

- Remove the unnecessary `.contiguous()` call from the NumPy input path in `to_image`.
- Add a regression test covering channels-last memory format.

## Testing

- `TestToImage`: 11 passed
- Regression test for channels-last memory format: passed
- `git diff --check`: passed

The full `test_transforms_v2.py` suite was also run. The unrelated failures were due to the local build not having libjpeg support enabled.

Fixes #9547

### CODE REVIEW

**Review by fallenmi — APPROVED** [review]
Reviewed exact head `d9b44cf1e1f406df2b1f5798926afba4c3e54c51` against exact base `ac8d215f7d45d6601451b62e9f81622dac8aa0b4`.

Removing the forced contiguous copy preserves the ordinary NumPy HWC storage as a CHW-shaped tensor with channels-last strides. In an independent matrix, standard RGB, RGBA, and float32 HWC arrays become channels-last after adding the batch dimension on this head, while base converts them to contiguous channels-first storage.

I also exercised eight NumPy layouts—including grayscale, row-padded ROIs, strided channels/columns, Fortran order, and float32—through `to_image`, horizontal flip, resize, dtype conversion, and normalization. Base, head, and the current GitHub merge produced identical shapes and value digests for every applicable operation; the head preserves each input's actual strides rather than changing values. The focused `TestToImage` slice passes 11/11 on head and merge (10/10 on base), `git diff --check` is clean, and the current merge tree is byte-identical to the reviewed head.

GitHub currently exposes only the green CLA/import/internal-only checks for this PR, not the normal public test matrix; the exact local functional coverage above is therefore the basis of this approval. I found no blocking issue.

AI disclosure: I used OpenAI Codex to inspect the exact revisions and repository policy, construct and run the stride/layout and downstream-transform matrix, execute the focused tests, and draft this review. I verified the results and conclusion.
